FrEee/Game/Objects/Civilization/Cargo.cs

```diff
@@ -209,7 +209,7 @@ private int TakePopulationDamage(Hit hit, int damage, PRNG? dice = null)
 					break; // no more population
 				double popHPPerPerson = Mod.Current.Settings.PopulationHitpoints;
 				// TODO - don't ceiling the popKilled, just stack it up
-				int popKilled = (int)Math.Ceiling(hit.Shot.DamageType.PopulationDamage.Evaluate(hit.Shot) / popHPPerPerson);
+				int popKilled = (int)Math.Ceiling(hit.Shot?.DamageType.PopulationDamage.Evaluate(hit.Shot) ?? 0 / popHPPerPerson);

```

Comment:
Yeah, shots with no damage type don't make much sense. And is `??` precedent over `/`? We don't want to default to `0 / popHPPerPerson` if the damage type is null!

Reply:
I will add some parenthesis to make sure that the precedence is right. It seems that `Hit` class has a constructor that doesn't take `Shot` as a parameter (and leaves it as null), but sets damage value and type; shouldn't this be taken into account, somehow?

My reasoning for the 0 damage was that before, if the `hit` variable had a null for `Shot` property, the entire thing would throw NullReferenceException anyway when trying to access `DamageType` property of the `Shot`; setting it to 0 would mean less exceptions. There might be better ways (check for null `Shot` before and throw hand-crafted exception, or log some kind of warning and still return 0?), but it would probably be best if you decided which to take.

It seems that the Shot-less constructor is used in only one place (CommonExtensions->TakeDamage(this IDamageable d, DamageType dt, int damage, PRNG dice = null) ); maybe it can be replaced with some other method of dealing damage, or it could craft a synthetic Shot to pass to the full constructor? That way it can be ensured that the Shot is not-null and the entire problem goes away.